qa/workunits/rbd/rbd_mirror_snapshot.sh

```diff
@@ -410,6 +410,52 @@ for i in ${image2} ${image4}; do
     remove_image_retry ${CLUSTER2} ${POOL} ${i}
 done
 
+testlog "TEST: request image resync when remote is not primary"
+test_resync_image=test_resync_image
+create_image_and_enable_mirror ${CLUSTER2} ${POOL} ${test_resync_image} ${RBD_MIRROR_MODE}
+write_image ${CLUSTER2} ${POOL} ${test_resync_image} 100
+wait_for_image_replay_stopped ${CLUSTER2} ${POOL} ${test_resync_image}
+wait_for_image_replay_started ${CLUSTER1} ${POOL} ${test_resync_image}
+wait_for_replay_complete ${CLUSTER1} ${CLUSTER2} ${POOL} ${POOL} ${test_resync_image}
```

Comment:
Namespace remapping (`--remote-namespace` option to `rbd mirror pool enable` command) isn't supported in reef and so `wait_for_replay_complete()` doesn't have the additional `remote_pool` parameter.  As is, this would fail on an attempt to take a snapshot on a non-existing image named `mirror` instead of `test_resync_image` same as in https://github.com/ceph/ceph/pull/64739#discussion_r2369031811.

Need to adjust how some of the helpers are called (`wait_for_replay_complete()`, `compare_images()`, etc) and run the script against upstream reef.